ttyplot.c

```diff
@@ -282,7 +282,11 @@ static void paint_plot(void) {
 // (Related: https://stackoverflow.com/q/62315082)
 static void signal_handler(int signum) {
     const unsigned char signal_number = (unsigned char) signum;  // signum is either 2 (SIGINT) or 28 (SIGWINCH)
-    write(signal_write_fd, &signal_number, 1);
+    while (true) {
+        const ssize_t write_res = write(signal_write_fd, &signal_number, 1);
+        if ((write_res == 1) || (errno != EINTR))
+            break;
+    }
```

Comment:
Here, a `do...while` seems more intuitive to me than breaking out of an infinite loop.
```suggestion
    ssize_t write_res;
    do
        write_res = write(signal_write_fd, &signal_number, 1);
    while (write_res == -1 && errno == EINTR);
```

Reply:
@edgar-bonet it's a question of style and I don't like the need for moving the variable outside the loop personally, but since it's mostly a style preference, we can do your style, no problem. Let me try to make GitHub produce a commits from this and then turn them into squashes with you as the co-author. Let's see if that works out.